ui/appui-react/src/appui-react/pickers/ViewSelector.tsx

```diff
@@ -232,9 +241,16 @@ export class ViewSelector extends React.Component<ViewSelectorProps, ViewSelecto
         else if (this.state.showUnknown)
           unknown.push(viewItem);
       });
+
+      if(this._searchInput.length > 0){
+        views3dFiltered = views3d.filter((view) => view.name!.toLowerCase().includes(this._searchInput.toLowerCase()));
+        views2dFiltered = views2d.filter((view) => view.name!.toLowerCase().includes(this._searchInput.toLowerCase()));
+        sheetsFiltered = sheets.filter((view) => view.name!.toLowerCase().includes(this._searchInput.toLowerCase()));
+        unknownFiltered = unknown.filter((view) => view.name!. toLowerCase().includes(this._searchInput.toLowerCase()));
+      }
     }
 
-    this.setStateContainers(views3d, views2d, sheets, unknown);
+    this.setStateContainers(views3d, views2d, sheets, unknown, views3dFiltered!, views2dFiltered!, sheetsFiltered!, unknownFiltered!);
```

Comment:
I doubt you need to have the `!` here, `undefined` is expected by the code anyway.

Reply:
The filtered view variables need the `!` because the first time `loadViews()` is called, they will be declared but not assigned anything as no search has been inputed yet.

Reply:
Since it is expected please change the type of the `let *Filtered` to `ListItem[] | undefined`, this way, it will not complain that it is used before being initialized (and will clarify that `undefined` is a valid state of these variables)